packages/main/src/TableSelection.ts

```diff
@@ -204,6 +217,181 @@ class TableSelection extends UI5Element implements ITableFeature {
 		this._table.headerRow[0]._invalidate++;
 		this._table.rows.forEach(row => row._invalidate++);
 	}
+
+	_onkeydown(e: KeyboardEvent) {
+		if (!this.isMultiSelect() || !this._table || !e.shiftKey) {
+			return;
+		}
+
+		const focusedElement = getActiveElement(); // Assumption: The focused element is always the "next" row after navigation.
+
+		if (focusedElement === null) {
+			this._stopRangeSelection();
+			return;
+		}
+
+		if (!(isInstanceOfTableRow(focusedElement)) && !this._rangeSelection?.isMouse) {
```

Comment:
it seems like additional parentheses is not necessary or?

Reply:
if the row is selected and then navigate with arrow right to the cell and the pressing shift + down should select the next row or not?